src/json.c

```diff
@@ -0,0 +1,666 @@
+/**
+ * @file
+ * JSON <-> CBOR conversion
+ */
+
+/*
+ *******************************************************************
+ *
+ * Copyright 2018 Intel Corporation All rights reserved.
+ *
+ *-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+ */
+
+#include <ctype.h>
+#include <safe_lib.h>
+#include <string.h>
+#include <stdlib.h>
+#include <dps/private/cbor.h>
+#include <dps/dbg.h>
+#include <dps/json.h>
+
+#define MAX_INPUT_STRING_LEN  RSIZE_MAX_STR
```

Comment:
We may want to use (RSIZE_MAX_STR) - 1 here.  In all the other #define MAX cases we check if the value is <= RSIZE_MAX_STR.  Is there a test for values of input string around the max?

Reply:
Will fix